controllers/service_controller.go

```diff
@@ -106,10 +106,8 @@ func (r *ServiceReconciler) translateService(ctx context.Context,
 			Name: identity.Name,
 		}
 
-		authCred := &auth_credentials.AuthCredential{
-			KeySelector: identity.Credentials.KeySelector,
-			In:          identity.Credentials.In,
-		} // TODO: prepare for when missing credentials field
+		// TODO: prepare for when missing credentials field
+		authCred := auth_credentials.NewAuthCredential(identity.Credentials.KeySelector, identity.Credentials.In)
```

Comment:
I think the problem is that `identity.Credentials` can be `nil`, no?

Reply:
Even if the struct `Identity` has `	Credentials Credentials 'json:"credentials,omitempty"'` the Crendentials will never be `nil`, since in Golang structs don't have a zero value (pointers do).

I'll remove the TODO